### PR TITLE
Upgrade from log4j to log4j2

### PR DESCRIPTION
**Pull request by ns1123**
This PR prepares for Java 11 by upgrading log4j to log4j2

### CODE REVIEW

File: webui/npm-shrinkwrap.json [hunk 1]
```diff
@@ -2055,36 +2055,11 @@
         "lodash": "^4.17.4"
       }
     },
-    "@nodelib/fs.scandir": {
```
Comment (ZacBlanco) [inline webui/npm-shrinkwrap.json]: please undo the changes to this file

File: underfs/pom.xml [hunk 2]
```diff
@@ -51,8 +51,13 @@
...
     <dependency>
-      <groupId>log4j</groupId>
-      <artifactId>log4j</artifactId>
+      <groupId>org.apache.logging.log4j</groupId>
+      <artifactId>log4j-api</artifactId>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.logging.log4j</groupId>
+      <artifactId>log4j-core</artifactId>
```
Comment (ZacBlanco) [inline underfs/pom.xml:60]: Do we need both of these?

I would look into first upgrading slf4j to the latest, and then utilize the same log4j version in the slf4j log4j impl

Comment (ZacBlanco) [reply underfs/pom.xml:60]: I _think_ we'll need to replace the slf4j-log4j12 with the newer log4j-slf4j-impl https://mvnrepository.com/artifact/org.apache.logging.log4j/log4j-slf4j-impl/2.13.3

Comment (ns1123) [reply underfs/pom.xml:60]: Done

**Comment by yuzhu** [comment]
can we not delete npm-shrinkwrap.json and keep its content as it was?

**Comment by yuzhu** [comment]
@ns1123 did you try ospublish tarball?

File: shaded/client/pom.xml [hunk 3]
```diff
@@ -63,8 +63,14 @@
     </dependency>
     <!-- Move log4j to optional, since it is mainly used in test-->
     <dependency>
-      <groupId>log4j</groupId>
-      <artifactId>log4j</artifactId>
+      <groupId>org.apache.logging.log4j</groupId>
+      <artifactId>log4j-api</artifactId>
+      <scope>runtime</scope>
+      <optional>true</optional>
```
Comment (ZacBlanco) [inline shaded/client/pom.xml:69]: why is this optional/runtime?

Comment (ns1123) [reply shaded/client/pom.xml:69]: Because I'm a copy/paste noob

**Comment by yuzhu** [comment]
alluxio-bot, merge this please
